src/cgame/cg_players.cpp

```diff
@@ -3488,31 +3488,72 @@ void CG_PlayerDisconnect( vec3_t org )
 
 centity_t *CG_GetLocation( vec3_t origin )
 {
-	float     bestlen, len;
+	size_t activeCount = cg_highestActiveEntity - MAX_CLIENTS;
```

Comment:
This is off by 1 since cg_highestActiveEntity is an inclusive bound. But anyway the variable is not really needed; I suggest the following rewrite:

```
std::vector<std::array<float, 3>> positions;
std::vector<int> entities;

for ( size_t i = MAX_CLIENTS; i <= cg_highestActiveEntity; i++ )
{
    centity_t *eloc = &cg_entities[ num ];

    if ( !eloc->valid || eloc->currentState.eType != entityType_t::ET_LOCATION )
    {
        continue;
    }

    std::array<float, 3> pos;
    VectorCopy( eloc->lerpOrigin, pos );
    positions.push_back( pos );
    entities.push_back( i );
}
```

If you care about the vectors not being reallocated you can call `reserve(MAX_LOCATIONS)` on them.

Reply:
Done.